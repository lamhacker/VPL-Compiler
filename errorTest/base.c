#include "base.h"

float* mallocVec(int num)
{
    float* vec = malloc(sizeof(float) * num + SSE_ALIGN);
	vec = (float *) align(vec, SSE_ALIGN);
	return vec;
}

void freeVec(float* vec)
{
    free(vec);
}

void setValue(float* vec, float val, int num)
{
    for (int i = 0; i < num; i++)
    {
        vec[i] = val;
    }
}

void printVec(char* rep, float* vec, int num)
{
	printf("%s: ", rep);
    for (int i = 0; i < num; i++)
    {   
        printf("%f ", vec[i]);
    }
	printf("\n");
}


