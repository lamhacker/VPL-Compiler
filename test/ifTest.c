#include "base.h"
#define NUM (4)

extern void ifOne(long, float*);
extern void ifTwo(long, float*);
extern void ifThree(long, float*);
extern void ifFour(long, float*);
extern void ifFive(long, float*);
extern void ifSix(long, float*, float*);
extern void ifSeven(long, float*, float*);
extern void ifEight(long, float*);

int main(void)
{
    float* a = mallocVec(NUM);
	float* b = mallocVec(NUM);
	
	printf("ifOne: \n");
    setValue(a, 2, NUM);
    ifOne(NUM, a);
	printVec("a", a, NUM);
	
	printf("ifTwo: \n");
    setValue(a, 11, NUM);
    ifTwo(NUM, a);
	printVec("a", a, NUM);
	
	printf("ifThree: \n");
    setValue(a, 13, NUM);
    ifThree(NUM, a);
	printVec("a", a, NUM);
	
	printf("ifFour: \n");
    setValue(a, 16, NUM);
    ifFour(NUM, a);
	printVec("a", a, NUM);
	
	printf("ifFive: \n");
    setValue(a, 17, NUM);
    ifFive(NUM, a);
	printVec("a", a, NUM);
	
	printf("ifSix: \n");
    setValue(a, 6, NUM);
	setValue(b, 20, NUM);
    ifSix(NUM, a, b);
	printVec("a", a, NUM);
	printVec("b", b, NUM);
	
	printf("ifSeven: \n");
	setValue(a, 1, NUM);
	setValue(b, 20, NUM);
	ifSeven(NUM, a, b);
	printVec("a", a, NUM);
	printVec("b", b, NUM);
	
	printf("ifEight: \n");
    setValue(a, 4, NUM);
    ifEight(NUM, a);
	printVec("a", a, NUM);

	freeVec(a);
	freeVec(b);
}
