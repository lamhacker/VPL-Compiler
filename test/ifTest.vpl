func ifOne(a)
	if a < 10 then
		a = 4;
	endif
end

func ifTwo(a)
	if a >= 10 then
		a = 5;
	endif
end

func ifThree(a)
	if a >= 10 then
		a = 4;
		if a < 20 then
			a = 3;
		endif
	endif
end

func ifFour(a)
	if a >= 10 then
		if min(a, 1) < 20 then
			if a >= 15 then
				a = 3;
			endif
		endif
	endif
end

func ifFive(a)
	if a >= 10 then
		if min(a, 1) < 20 then
			if a >= 15 then
				a = 3;
			endif
		endif
	endif ;
	if add(a, 5) >= 10 then
		a = 1;
	endif
end

func ifSix(a, b)
	if mult(a, 2) >= 10 then
		a = 5;
	endif;
  	if div(b, 10) < 20 then
    	b = 10;
  	endif
end

func ifSeven(a, b)
	if mult(a, b) < 100 then
		a = add(a, 1);
		b = add(b, 5);
	endif
end

func ifEight(a)
	if 1 < 2 then
	endif
end
