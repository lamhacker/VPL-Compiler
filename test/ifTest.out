ifOne: 
a: 4.000000 4.000000 4.000000 4.000000 
ifTwo: 
a: 5.000000 5.000000 5.000000 5.000000 
ifThree: 
a: 3.000000 3.000000 3.000000 3.000000 
ifFour: 
a: 3.000000 3.000000 3.000000 3.000000 
ifFive: 
a: 1.000000 1.000000 1.000000 1.000000 
ifSix: 
a: 5.000000 5.000000 5.000000 5.000000 
b: 10.000000 10.000000 10.000000 10.000000 
ifSeven: 
a: 2.000000 2.000000 2.000000 2.000000 
b: 25.000000 25.000000 25.000000 25.000000 
ifEight: 
a: 4.000000 4.000000 4.000000 4.000000 
